awsgi/__init__.py

```diff
@@ -16,29 +17,49 @@ def convert_str(s):
         return s
 
 
-def response(app, event, context):
-    sr = StartResponse()
+def convert_b46(s):
+    return b64encode(s).decode("ascii")
+
+
+def response(app, event, context, base64_content_types=None):
+    sr = StartResponse(base64_content_types=base64_content_types)
     output = app(environ(event, context), sr)
     return sr.response(output)
 
 
 class StartResponse:
-    def __init__(self):
+    def __init__(self, base64_content_types=None):
+        """
+        Args:
+            base64_content_types (set): Set of HTTP Content-Types which should
+            return a base64 encoded body. Enables returning binary content from
+            API Gateway.
+        """
         self.status = 500
         self.headers = []
         self.body = StringIO()
+        self.base64_content_types = (
+            set(base64_content_types) if base64_content_types else set()
+        )
 
     def __call__(self, status, headers, exc_info=None):
         self.status = status.split()[0]
         self.headers[:] = headers
         return self.body.write
 
     def response(self, output):
+        headers = dict(self.headers)
+        is_b64 = headers.get("Content-Type") in self.base64_content_types
+        converted_output = (
```

Comment:
I think that traditional `if: ... else:` is easier to read and it's hardly less concise in this case.

Reply:
Agreed :+1: